javascripts/helpers.js

```diff
@@ -0,0 +1,46 @@
+// allows us to compare an array of arrays
+// by turning the inner arrays into string, e.g. ["r3, c3", "r2, c0"]
+function stringify(array) {
+  var stringifiedArray = [];
+
+  for (var i = 0; i < array.length; i++) {
+    stringifiedArray.push(array[i].join());
+  }
+
+  return stringifiedArray;
+}
+
+// returns an array of the occupied spaces, e.g. ["r2", "c0"]
+function occupiedSpaces() {
+  var tiles = $('.tile');
+
+  var takenSpaces = [];
+  for (var i = 0; i < tiles.length; i++) {
+    var row = tiles[i].getAttribute("data-row");
+    var col = tiles[i].getAttribute("data-col");
+    takenSpaces.push([row, col]);
+  }
+
+  return takenSpaces;
+}
+
+// returns an array of open space locations, e.g. ["r1", "c3"]
+function openSpaces() {
+  var takenSpaces = stringify(occupiedSpaces());
```

Comment:
Turning a data structure into a string and then parsing it back out is clever, but generally a brittle thing to do. The string form can change over time and then break your code. It's also harder to tell *why* the code is broken when it stops working because you don't know what the original string representation is.